tracee-ebpf/tracee/tracee.go

```diff
@@ -286,13 +288,17 @@ func New(cfg Config) (*Tracee, error) {
 		setEssential(SecuritySocketBindEventID)
 	}
 
-	systemUptime, _ := Uptime()
-	systemBootTime := time.Now().UnixNano() - systemUptime*1000*1000*1000
+	// Tracee bpf code uses monotonic clock as event timestamp.
+	// Get current monotonic clock so we can calculate event timestamps relative to it.
+	monotonicStartTime, _ := GetClockTime(CLOCK_MONOTONIC)
+	timeSinceBoot, _ := GetClockTime(CLOCK_BOOTTIME)
+	bootWallTime := time.Now().UnixNano() - timeSinceBoot
 
 	// create tracee
 	t := &Tracee{
-		config:   cfg,
-		bootTime: uint64(systemBootTime),
+		config:    cfg,
+		startTime: uint64(monotonicStartTime),
```

Comment:
CLOCK_MONOTONIC is a clock source indicating the monotonic time elapsed since boot, not including time spent in suspend (for this there is CLOCK_BOOTTIME). We take the value of this clock when tracee is started (here), then we can use it as a reference for the other events to get the monotonic time elapsed since tracee was started.